### PR TITLE
QgisFormControl tests

### PR DESCRIPTION
**Pull request by alagroy-42**
Tests for Lizmap\Form\QgisFormControl.

It's not a lot but despite the refactorisation, most of the methods are either useless to test or still not testable.

### CODE REVIEW

File: lizmap/modules/lizmap/lib/App/JelixContext.php [hunk 1]
```diff
@@ -277,4 +278,14 @@ public function getFormPath()
     {
         return \jApp::tempPath('lizmap-forms');
     }
+
+    public function getFormPath()
+    {
+        return realpath(__DIR__.'/../../').'/forms/';
+    }
+
+    public function getFormPath()
```
Comment (laurentj) [inline lizmap/modules/lizmap/lib/App/JelixContext.php:287]: method duplicated

File: lizmap/modules/lizmap/lib/App/JelixContext.php [hunk 2]
```diff
@@ -277,4 +278,14 @@ public function getFormPath()
     {
         return \jApp::tempPath('lizmap-forms');
     }
+
+    public function getFormPath()
```
Comment (laurentj) [inline lizmap/modules/lizmap/lib/App/JelixContext.php:282]: duplicated method

File: lizmap/modules/lizmap/lib/Form/QgisForm.php [hunk 3]
```diff
@@ -104,6 +104,9 @@ public function __construct($layer, $form, $featureId, $loginFilteredOverride, A
         $toSetReadOnly = array();
         $formPath = $this->appContext->getFormPath();
         $json = file_get_contents($formPath.$layer->getProject()->getKey().'.'.$layer->getId().'.form.json');
+        if (!$json) {
+            throw new \Exception('Can\'t read the Json form file, try to clear you cache and reload the page');
```
Comment (laurentj) [inline lizmap/modules/lizmap/lib/Form/QgisForm.php:108]: s/you/your

File: lizmap/modules/lizmap/lib/Form/QgisForm.php [hunk 4]
```diff
@@ -546,6 +541,7 @@ public function saveToDb($feature = null)
             throw new \Exception('Save to database can\'t be done for the layer "'.$this->layer->getName().'"!');
         }
 
+        $cnx = $this->layer->getDatasourceConnection();
```
Comment (laurentj) [inline lizmap/modules/lizmap/lib/Form/QgisForm.php:544]: this variable is not used below. You should not add this line of code.

**Review by laurentj — CHANGES_REQUESTED** [review]

File: lizmap/modules/lizmap/lib/Form/QgisForm.php (hunk 4 above)
Comment (alagroy-42) [reply lizmap/modules/lizmap/lib/Form/QgisForm.php:544]: It is [there](https://github.com/alagroy-42/lizmap-web-client/blob/2d83af54836fae08eb0770b2e9e44d7b2aaca0b1/lizmap/modules/lizmap/lib/Form/QgisForm.php#L575)

Comment (laurentj) [reply lizmap/modules/lizmap/lib/Form/QgisForm.php:544]: Sorry, I looked into the `check()` method, where `$cnx` is unused.

**Review by laurentj — APPROVED** [review]

**Pull request merged**